### ISSUE TITLE
Galicaster incorrectly sets the MediaPackage ID of the scheduled recordings

### ISSUE THREAD

**Issue by sinhote** [teltek/Galicaster#173]
Galicaster uses the `UID` field in the calendar events exported by Matterhorn/Opencast as the MediaPackage ID, when it creates the events in the Repository. That is however incorrect, as you can see here:

    BEGIN:VEVENT
    DTSTAMP:20160428T160136Z
    DTSTART:20160429T075500Z
    DTEND:20160429T094000Z
    SUMMARY:Einführung in das Kirchenrecht (G) (I-IV) - SS 16 - 2016-04-29
    UID:3371298
    ORGANIZER;CN=Prof. Dr. Stefan Muckel
    LOCATION:GCMobile-rrzkca-hsa1
    RELATED-TO:1ada27b6-5c05-4447-a89a-7777545bf1a5

The MediaPackage ID appears in the field `RELATED-TO`, while the number in the `UID` field is just the Workflow ID. 

This causes problems when, for instance, a recording has to be uploaded again. The first time, because Opencast is waiting for the scheduled workflow to capture, the MP is correctly assigned (i.e.  the MediaPackage ID that comes from the capture agent is ignored), but the second time, the same video will be assigned a different MediaPackage ID.

**Comment by smarquard** [comment]
Actually RELATED-TO: contains the Series ID, not the mediapackage ID.

**Comment by sinhote** [comment]
@smarquard 

Oh, sheet! Then there's no way to get the MP ID?

**Comment by smarquard** [comment]
I think the mediapackage ID might only exist after ingest? I'm not sure how this works in 2.x either, as I don't think there's an initial workflow for a scheduled event.

**Comment by rubenrua** [comment]
OMG!!

I thought that the MP ID can be obtained from the attached `episode.xml` in the ical. But in 2.x  the `dcterms:identifier` is not he MP ID, I have to figure out what is this number.

![oc2](https://cloud.githubusercontent.com/assets/195745/14910490/244151d8-0dee-11e6-8c5f-f8f46af2a12c.png)

```xml
<?xml version="1.0" encoding="UTF-8"?>
<dublincore xmlns="http://www.opencastproject.org/xsd/1.0/dublincore/" xmlns:dcterms="http://purl.org/dc/terms/" xmlns:oc="http://www.opencastproject.org/matterhorn/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <dcterms:temporal xsi:type="dcterms:Period">start=2016-09-05T23:01:00Z; end=2016-09-06T02:04:00Z; scheme=W3C-DTF;</dcterms:temporal>
  <dcterms:title>Test #173</dcterms:title>
  <dcterms:identifier>5017552835992547707</dcterms:identifier>
  <dcterms:spatial>GCMobile-bonito</dcterms:spatial>
  <oc:duration>10980000</oc:duration>
  <dcterms:subject>GC test</dcterms:subject>
  <dcterms:created>2016-04-29T07:16:51Z</dcterms:created>
</dublincore>

```

[get_info_from_ical.py](https://github.com/teltek/Galicaster/files/241923/ical.txt)

**Comment by spiraleddy** [comment]
Ruben,

At least up through MH/OC v1.63, the Scheduler Service iCal's version of dcterms:identifier has always been the workflow Id, aka first workflow job Id, aka event id. 

Beware that the " <dcterms:identifier>5017552835992547707</dcterms:identifier>" is a field set directly by the scheduler service just for the CA/iCal version of the episode dublin core catalog. The actual Episode dublincore catalog submitted to the scheduler service, and existing within  the OC system, does not have that value set in it's identifier field (at least up through v1.63).

- Karen

Karen Dolan
Harvard University DCE

**Comment by smarquard** [comment]
There is some question (in my mind at least) about whether Opencast 2.x is doing the right thing here - related to https://opencast.jira.com/browse/MH-11995

The UID field in the iCal feed for the CA is the scheduler event id (mh_scheduled_event.id) and it should possibly be the mediapackage ID (which OC 2.x regards as the "event id"), mh_scheduled_event.mediapackage_id, which is what's shown in the UID of the event details in the screenshot above.

In OC 2.x, there's no workflow associated with a scheduled event until the mediapackage is ingested.

**Comment by sinhote** [comment]
Well, we could say this is actually a non-bug (in Galicaster, at least), because I misinterpreted what the `dcterms:identifier` really represented. As Karen pointed out, it is not the series ID, but the event ID (i.e. the id of the first job in the workflow, i.e. the `START_WORKFLOW` job).

This is obviously misguiding, because the identifier is supposed to be the **mediapackage identifier**, while the event ID has little to no meaning outside the system's internals. However, Galicaster is doing its best here, because the mediapackage ID does not get sent in the iCal --only the event ID. The reasons are, as almost everything weird or confusing in Opencast, historical: at first the mediapackage would only be created upon ingestion, so before that there was no mediapackage ID that could actually be sent to the capture agents. Note that the ingest service is practically untouched since version 1.0.

In short, no Galicaster bug, but perhaps an Opencast bug, which I think should not be dealt with here. At least, not until Opencast redesigns this ID mess into something a little less confusing.

**Comment by smarquard** [comment]
Agreed (that it's not a Galicaster bug)
